plugins/module_utils/csv.py

```diff
@@ -55,12 +55,17 @@ class unix_dialect(csv.Dialect):
 
 
 def read_csv(data, dialect, fieldnames=None):
-
+    BOM = u'\ufeff'
     data = to_native(data, errors='surrogate_or_strict')
 
     if PY3:
+        if data.startswith(BOM):
+            data = data[1:]
         fake_fh = StringIO(data)
     else:
+        # type(data) == bytes in Py2, and encoded BOM is 3 bytes long
+        if data.startswith(BOM.encode('utf-8')):
+            data = data[3:]
         fake_fh = BytesIO(data)
```

Comment:
How about doing `BOM = to_native(u'\ufeff')` and then `if data.startswith(BOM): data = data[len(BOM):]` before the `if PY3`? That's easier to understand IMO and it avoids magic numbers.

Reply:
much better indeed